src/compiler/program.ts

```diff
@@ -1560,6 +1560,7 @@ namespace ts {
         }
 
         function getDiagnosticsProducingTypeChecker() {
+            Debug.assert(!options.listFilesOnly || !!options.extendedDiagnostics);

```

Comment:
@RyanCavanaugh is there something softer than an assert that I could/should use here?  Asserting is great locally, but telemetry would suffice on a user's box.

Reply:
Removed from 3.7.  Will be revisited in 3.8 (possibly as a telemetry-only assert).